src/Channels/ReadableBuffer.cs

```diff
@@ -82,10 +82,11 @@ private ReadableBuffer(ref ReadableBuffer buffer)
             begin = new ReadCursor(segmentHead);
             end = new ReadCursor(segmentTail, segmentTail.End);
 
+            begin.TryGetBuffer(end, out _span, out begin);
+
             _start = begin;
             _end = end;
             _isOwner = true;
-            _span = buffer._span;
```

Comment:
FYI, this change was needed for Preserve to actually work properly when `IBuffer.Preserve` returns a **different** Buffer object